### ISSUE TITLE
NPCs touch then cancel dissassembing piles due to low morale with no warning

### ISSUE THREAD

**Issue by kishitoku** [CleverRaven/Cataclysm-DDA#64014]
### Describe the bug

NPCs will pick up an item from a deconstruct pile, turn it into a 0% deconstructed object and set it down, then declare their task complete and abandon the entire deconstruct order, in some situations.

### Attach save file

[Fortville-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/10905060/Fortville-trimmed.tar.gz)

### Steps to reproduce

* Create a new world with a survivor starting in the shelter
* Acquire the loyalty of the NPC through a dialogue check or other methods
* Spawn in a pile containing a fridge, two game consoles, and four sheaths
* Make a deconstruct zone over the pile
* Spawn in and trade to the NPC a charged soldering iron, multitool, scissors, and bone awl
* Order the NPC to deconstruct
* The behavior occurs

### Expected behavior

The NPC should deconstruct the items - they seem to have everything they need. If for some reason they can't find an appropriate tool, they should skip the item and move on to the next one and generate the appropriate dialogue, not abandon the entire deconstruct stack with no additional feedback.

### Screenshots

![Untitled](https://user-images.githubusercontent.com/74622599/223312367-155c5a95-04ba-4529-a9fc-f48e393f246a.png)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.18363.2037 (1909)
- Game Version: d6ec466 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

The issue doesn't seem to be the fridge; moving it out of the pile doesn't stop the behavior from occurring. In the main game where I noticed this I had tons of tools lying around, more than enough to deconstruct the items myself.

I seem to be able to deconstruct a sheath and game console just fine when I take the tools from them as a new survivor, so it's not that the deconstruction can't happen; it just isn't being done.

**Comment by Ryan7251** [comment]
I hate to tell you this but I'm not having this issue?

**Comment by Karol1223** [comment]
I can confirm this has happened to me about a week ago. Hell, I even know the culprit.

It's the damn game consoles.

For whatever reason, they are the only recipe in the game that, when reversed (i.e. disassembling, because it is defined by a reversible recipe) needs *morale* to progress with the disassembly. This means that it is the only item in the game that you cannot disassemble when you are sad and it appears that 9/10 times NPCs cannot disassemble it either.

**Comment by kishitoku** [comment]
Okay, that's really helpful news to hear.  Unfortunately, in my crafting pile on my main save, it seems like removing the game consoles from the disassembly pile somehow doesn't fix the issue. Even removing everything but all the 0% progress sheaths, the NPC will just pick them up and set them down.

So is it possible that encountering this issue has somehow bugged my NPC to not function even after the video games are removed? Or is it because they
[West Middlesex-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/10911200/West.Middlesex-trimmed.tar.gz)
 were started and not completed? Or maybe because there were video games in there at one point the bug has somehow infected them?

Attaching main game save.

**Comment by kishitoku** [comment]
Update, I have reversed the bugged pile of stuff in my main save! Investigating further off of karol's info, I took my NPC and read several chapters of Dune to him to cheer him up. After I was done he was once again able to disassemble sheaths.

So I still don't know if the consoles screwed him up (I never encountered this low morale issue before I put a video game in the pile, and I had him working for like a season with no issue), but I at least know a temporary fix.

If it is intended that an NPC can be too sad to disassemble, there needs to be some feedback! We need a message saying why the NPC quit their work, or at least an easy way to check their morale. If this isn't intended then it should be fixed.
